### ISSUE TITLE
Broken cloudflare workers sourcemaps 

### ISSUE THREAD

**Issue by tiagoad** [remix-run/remix#2464]
### What version of Remix are you using?

1.3.2

### Steps to Reproduce

- Build a project targeting Cloudflare workers with `remix build --sourcemap`
- Run `npx source-map-explorer build/index.js`

### Expected Behavior

Sourcemap is correctly parsed

### Actual Behavior

Output:
```
build/index.js
  Your source map refers to generated column 1 on line 20, but the source only contains 0 column(s) on that line.
  Check that you are using the correct source map.
```

**Comment by mcansh** [comment]
doesn't appear that `remix build --sourcemap` generates server source maps, but just client ones

**Comment by tiagoad** [comment]
> doesn't appear that `remix build --sourcemap` generates server source maps, but just client ones

It generates inline sourcemaps: https://github.com/remix-run/remix/blob/main/packages/remix-dev/compiler.ts#L452
If you look at the last line in your generated worker javascript file you'll see it starts with `//# sourceMappingURL=data:application/json;`

**Comment by kiliman** [comment]
My PR #2065 fixes this to use "linked" sourcemaps. It also rewrites route module references so breakpoint debugging works without any hacks.

**Comment by machour** [comment]
@tiagoad can you try again and tell us if it's fixed for you now?

There's also #3768 that have been reported after #2065 got merged, in case of problems, can you tell us if it affects you as well?

**Comment by machour** [comment]
Closing this as no response was provided from the original issue author. If you still feel like something is wrong on our side, please open a new issue using the latest remix version.

**Comment by lpsinger** [comment]
I am seeing the same error message with Architect (AWS Lambda) projects.
